src/main/scala/com/microsoft/ml/spark/core/utils/ClusterUtil.scala

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark.core.utils
+
+import java.net.InetAddress
+
+import org.apache.http.conn.util.InetAddressUtils
+import org.apache.spark.lightgbm.BlockManagerUtils
+import org.apache.spark.sql.Dataset
+import org.slf4j.Logger
+
+object ClusterUtil {
+  /** Get number of cores from dummy dataset for 1 executor.
+    * Note: all executors have same number of cores,
+    * and this is more reliable than getting value from conf.
+    * @param dataset The dataset containing the current spark session.
+    * @return The number of cores per executor.
+    */
+  def getNumCoresPerExecutor(dataset: Dataset[_], log: Logger): Int = {
```

Comment:
it looks like you are redefining this method twice, I think one of the files is a bad copy